feat: split optimize upgrade to camunda 8 & 7 versions (#4241)
camunda/camunda-docs: optimize/self-managed/optimize-deployment/migration-update/camunda-7/3.10-to-3.11.md

### Java

With this release, the minimum version of Java that Optimize supports is now Java 17. See the [Supported Environments]($docs$/reference/supported-environments) sections for more information on supported versions.

### Plugins

Optimize now runs with Spring Boot 3. As a result, some plugin interfaces have been updated accordingly. More specifically, the [Engine Rest Filter Plugin](./../../plugins/engine-rest-filter-plugin.md) and the [Single-Sign-On Plugin](./../../plugins/single-sign-on.md) now import jakarta dependencies. If you use these plugins and are updating from version 3.10.3 or earlier, you will need to adjust your implementation accordingly.

### Logging

With the change to Spring Boot 3, Optimize's logging configuration format has also been updated. If you are updating from version 3.10.3 or earlier, please review the updated `environment-logback.xml` to make sure your configuration is valid.

## Changes in behavior

### Collection Role Cleanup

> Prior to Optimize 8.3/3.11, Optimize has performed collection role cleanup after syncing identities with the engine. From

Reviewer: ```suggestion
Prior to Optimize 3.11, Optimize has performed collection role cleanup after syncing identities with the engine. From
```

> Optimize 8.3/3.11 onwards, this is now disabled by default. It can be reenabled by setting the

Reviewer: ```suggestion
Optimize 3.11 onwards, this is now disabled by default. It can be reenabled by setting the
```

`import.identitySync.collectionRoleCleanupEnabled` property value to `true`

### API behavior

> Before the 7.20/8.3 release, the Optimize API would accept requests when the URI contained a trailing slash (`/`). This is no longer the case, and requests containing a trailing slash will no longer be matched to the corresponding API path.

Reviewer: ```suggestion
Before the 3.11 release, the Optimize API would accept requests when the URI contained a trailing slash (`/`). This is no longer the case, and requests containing a trailing slash will no longer be matched to the corresponding API path.
```
